drivers/include/lis2dh12.h

```diff
@@ -131,76 +193,49 @@ enum{
-    LIS2DH12_INT_CFG_ZLIE = 0x10,   /**< enable Z low event */
-    LIS2DH12_INT_CFG_ZHIE = 0x20,   /**< enable Z high event */
-    LIS2DH12_INT_CFG_6D   = 0x40,   /**< enable 6-direction detection */
-    LIS2DH12_INT_CFG_AOI  = 0x80,   /**< and/or combination interrupt events */
-};
-
-/**
- * @brief   Interrupt type values
- */
-enum {
-    /* for interrupt 1 (CTRL_REG3) */
-    LIS2DH12_INT_TYPE_I1_OVERRUN = 0x02, /**< FIFO overrun interrupt on INT1 */
-    LIS2DH12_INT_TYPE_I1_WTM     = 0x04, /**< FIFO watermark inter. on INT1 */
-    LIS2DH12_INT_TYPE_I1_ZYXDA   = 0x10, /**< ZYXDA interrupt on INT1 */
-    LIS2DH12_INT_TYPE_I1_IA2     = 0x20, /**< IA2 interrupt on INT1 */
-    LIS2DH12_INT_TYPE_I1_IA1     = 0x40, /**< IA1 interrupt on INT1 */
-    LIS2DH12_INT_TYPE_I1_CLICK   = 0x80, /**< click interrupt on INT1 */
-
-    /* for interrupt 2 (CTRL_REG6) */
-    LIS2DH12_INT_TYPE_INT_POLARITY = 0x02, /**< INT1 and INT2 pin polarity */
-    LIS2DH12_INT_TYPE_I2_ACT       = 0x08, /**< enable activity interrupt on INT2 */
-    LIS2DH12_INT_TYPE_I2_BOOT      = 0x10, /**< enable boot on INT2 */
-    LIS2DH12_INT_TYPE_I2_IA2       = 0x20, /**< IA2 on INT2 */
-    LIS2DH12_INT_TYPE_I2_IA1       = 0x40, /**< IA1 on INT2 */
-    LIS2DH12_INT_TYPE_I2_CLICK     = 0x80, /**< click interrupt on INT2 */
-};
-
 /**
  * @brief    Parameter for interrupt configuration
  */
 typedef struct {
     uint8_t int_config;             /**< values for configuration */
-    uint8_t int_threshold:7;        /**< the threshold for triggering interrupt, threshold in range 0-127 */
-    uint8_t int_duration:7;         /**< time between two interrupts ODR section in CTRL_REG1, duration in range 0-127 */
+    uint8_t int_threshold:7;        /**< the threshold for triggering interrupt,
+                                        threshold in range 0-127 */
+    uint8_t int_duration:7;         /**< time between two interrupts ODR section in CTRL_REG1,
+                                        duration in range 0-127 */
     uint8_t int_type;               /**< values for type of interrupts */
     gpio_cb_t cb;                   /**< the callback to execute */
     void *arg;                      /**< the callback argument */
 } lis2dh12_int_params_t;
 
 /**
- * @brief   Status of INT_SRC register
+ * @brief   LIS2DH12 FIFO data struct
  */
-#define LIS2DH12_INT_SRC_XL     (0x01)  /**< X low event has occurred */
-#define LIS2DH12_INT_SRC_XH     (0x02)  /**< X high event has occurred */
-#define LIS2DH12_INT_SRC_YL     (0x04)  /**< Y low event has occurred */
-#define LIS2DH12_INT_SRC_YH     (0x08)  /**< Y high event has occurred */
-#define LIS2DH12_INT_SRC_ZL     (0x10)  /**< Z low event has occurred */
-#define LIS2DH12_INT_SRC_ZH     (0x20)  /**< Z high event has occurred */
-#define LIS2DH12_INT_SRC_IA     (0x40)  /**< 1 if interrupt occurred */
-#endif /* MODULE_LIS2DH12_INT */
+typedef struct {
+    int16_t FIFO_data_X;    /**< X raw data in FIFO */
```

Comment:
```suggestion
    int16_t x;    /**< X raw data in FIFO */
```
no need to prefix struct members